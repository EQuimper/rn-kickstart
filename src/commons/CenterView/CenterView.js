// @flow

import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import { type StyleObj } from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

type P = {
  children: React.Node,
  style?: StyleObj,
};

function CenterView({ style, children }: P) {
  return <View style={[styles.root, style]}>{children}</View>;
}

const styles = StyleSheet.create({
  root: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default CenterView;
