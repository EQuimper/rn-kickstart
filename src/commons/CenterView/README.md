# CenterView

Putting element in the middle of is container. Take the full size of his parent, and put element in the middle of himself.