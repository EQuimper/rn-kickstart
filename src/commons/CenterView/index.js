// @flow

export { default as CenterView } from './CenterView';
