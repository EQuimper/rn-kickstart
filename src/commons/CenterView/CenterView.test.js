// @noflow

import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';

import CenterView from './CenterView';

describe('<CenterView />', () => {
  it('should renders correctly', () => {
    const tree = shallow(
      <CenterView>
        <Text>Hello World</Text>
      </CenterView>,
    );

    expect(tree).toMatchSnapshot();
  });
});
