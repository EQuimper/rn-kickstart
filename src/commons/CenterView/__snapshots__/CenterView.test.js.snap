// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CenterView /> should renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Hello World
  </Text>
</View>
`;
